packages/stateManager/tests/stateManager.test.ts

```diff
@@ -0,0 +1,274 @@
+import { describe, expect, test, vi } from "vitest";
+import { Level } from "level";
+import { StateManager } from "@massmarket/stateManager";
+import { testVectors } from "@massmarket/schema";
+import { MockClient } from "./mockClient";
+import { randomAddress, random32BytesHex } from "@massmarket/utils";
+
+export const zeroAddress = "0x0000000000000000000000000000000000000000";
+const db = new Level(`./tmpDb`, {
+  valueEncoding: "json",
+});
+db.clear();
+const listingStore = db.sublevel("listingStore", {
+  valueEncoding: "json",
+});
+const tagStore = db.sublevel("tagStore", {
+  valueEncoding: "json",
+});
+const shopManifestStore = db.sublevel("shopManifestStore", {
+  valueEncoding: "json",
+});
+const orderStore = db.sublevel("orderStore", {
+  valueEncoding: "json",
+});
+
+describe("Fill state manager with test vectors", async () => {
+  const client = new MockClient();
+  const stateManager = new StateManager(
+    // @ts-expect-error FIXME looking for client to be RelayClient
+    client,
+    listingStore,
+    tagStore,
+    shopManifestStore,
+    orderStore,
+  );
+  let r = 0;
+  client.on("shopManifest", (e) => {
+    console.log("event:shopManifest", e);
+    r = 1;
+  });
+  client.on("updateShopManifest", (e) => {
+    console.log("event:updateShopManifest", e);
+  });
+  await vi.waitUntil(async () => {
```

Comment:
trying to do something like this so we only move on to the tests once the event is received, but this waitUntil never resolves.